odl/tomo/data/mrc2014.py

```diff
@@ -0,0 +1,846 @@
+    def data_shape(self):
+        """Shape tuple of the data after considering `data_axis_order`.
+
+        If no header is available (i.e., before it has been initialized),
+        or any of the header entries ``'nx', 'ny', 'nz'`` is missing,
+        -1 is returned, which makes reshaping a no-op.
+        Otherwise, the returned shape is ``(nx, ny, nz)``, permuted
+        according to `data_axis_order`, precisely::
+
+            num_cells[i] == data_shape[data_axis_order[i]]
+        """
+        ncells = self.num_cells
+        if ncells == -1:
+            return ncells
+        else:
+            # Need to invert the permutation
+            return tuple(ncells[ax]
+                         for ax in np.argsort(self.data_axis_order))
+
+    @property
+    def data_dtype(self):
+        """Data type of the data block as determined from `header`.
+
+        If no header is available (i.e., before it has been initialized),
+        or the header entry ``'mode'`` is missing, the data type gained
+        from the ``dtype`` argument in the initializer is returned.
+        Otherwise, it is determined from ``mode``.
+        """
+        if not self.header:
+            return self._init_data_dtype
+        try:
+            mode = int(self.header['mode']['value'])
+        except KeyError:
+            return self._init_data_dtype
+        else:
+            try:
+                return MRC_MODE_TO_NPY_DTYPE[mode]
+            except KeyError:
+                raise ValueError('data mode {} not supported'.format(mode))
+
+    @property
+    def data_kind(self):
+        """String ``'volume'``, ``'projections'`` or ``'unknown'``.
+
+        The value is determined from the ``'ispg'`` header entry.
+        """
+        ispg = self.header['ispg']['value']
+        if ispg == 0:
+            return 'projections'
+        elif ispg == 1:
+            return 'volume'
+        else:
+            return 'unknown'
+
+    @property
+    def data_axis_order(self):
+        """Permutation of ``(0, 1, 2)``.
+
+        If no header is available, (i.e., before it has been initialized),
+        or one of the header entries ``'mapc', 'mapr', 'maps'`` is missing,
+        the identity permutation ``(1, 2, 3)`` is returned.
+        Otherwise, value is determined from the ``'mapc', 'mapr', 'maps'``
+        header entries.
+        """
+        if not self.header:
+            return (0, 1, 2)
+        try:
+            mapc = self.header['mapc']['value']
+            mapr = self.header['mapr']['value']
+            maps = self.header['maps']['value']
+        except KeyError:
+            return (0, 1, 2)
+        else:
+            return tuple(int(m) - 1 for m in (mapc, mapr, maps))
+
+    @property
+    def cell_sides_angstrom(self):
+        """Array of sizes of a unit cell in Angstroms.
+
+        The value is determined from the ``'cella'`` entry in `header`.
+        """
+        return (np.asarray(self.header['cella']['value'], dtype=float) /
```

Comment:
IMO the readability of
```python
        return np.asarray(self.header['cella']['value'],
                          dtype=float) / self.num_cells
```
is worse than
```python
        return (np.asarray(self.header['cella']['value'], dtype=float) /
                self.num_cells)
```
In the latter case, you split at the operator, in contrast to an arbitrary point in the function call as in the first case.
Third option, dunno:
```python
        return (np.asarray(self.header['cella']['value'],
                           dtype=float) /
                self.num_cells)
```
Kind of ugly?